Sources/DeepLearning/Layer.swift

```diff
@@ -51,13 +51,14 @@ public extension Layer {
 /// In typical uses, every layer in a model that has behavior which differs
 /// between training and inference shares an instance of ModeRef so it doesn't
 /// need to be toggled or threaded through in more than one place.
-public class ModeRef {
-    var training: Bool = true
+public class LearningPhaseIndicator {
+    public var training: Bool = true
+    public init() { }
 }
 
 /// A mutable, shareable reference to a tensor
 public class Parameter<T : TensorFlowScalar> {
```

Comment:
Nit: In non-stdlib code, not add a space before a colon in all cases (including the inheritance syntax).
```suggestion
public class Parameter<T: TensorFlowScalar> {
```